extensions/target-specific/chromium/blink/fragment-tree/fragment-tree.js

```diff
@@ -34,27 +50,7 @@ Loader.OnLoad(function() {
                             "Enable LayoutNG in chrome://flags or run with --enable-blink-features=LayoutNG");
                     return Promise.reject(errorMessage);
                 }
-                return BlinkHelpers.GetDocuments().then((documents) => {
-                    if (documents.length == 0) {
-                        var errorMessage = ErrorMessages.CreateErrorsList("No documents found.") +
-                            ErrorMessages.CreateErrorReasonsList(ErrorMessages.WrongDebuggee("the Chromium renderer process"),
-                                "The debuggee has been broken into prior to <i>g_frame_map</i> being populated.",
-                                ErrorMessages.SymbolsUnavailable) +
-                            "You may still specify a blink::NGPhysicalFragment explicitly.";
-                        return Promise.reject(errorMessage);
-                    }
-                    // We have to get the child of the root layout object (the LayoutView) because LayoutView never has a fragment.
-                    // All documents have a LayoutView, so we don't have to nullcheck that.
-                    let firstChildrenPromise = Promise.map(documents,
-                        (document) => document.F("node_layout_data_").f("layout_object_").vcast().f("children_.first_child_").vcast())
-                        .then((array) => array.filter((layout_object) => !layout_object.isNull()));
-                    return Promise.map(firstChildrenPromise,
-                        (layout_object) => layout_object.f("cached_layout_result_.ptr_.physical_fragment_.ptr_"));
-                }, (error) => {
-                    var errorMessage = ErrorMessages.CreateErrorsList(error) +
-                        ErrorMessages.CreateErrorReasonsList(ErrorMessages.WrongDebuggee("the Chromium renderer process"), ErrorMessages.SymbolsUnavailable);
-                    return Promise.reject(errorMessage);
-                });
+                return BlinkHelpers.GetRootLayoutObjects("blink::LayoutObject or blink::NGPhysicalFragment");

```

Comment:
Instead of explicitly writing out "or", it might be more elegant to let GetRootLayoutObjects take a variable number of parameters with the rest syntax (...), and then add the "or" when creating the error message.

Reply:
Done